library/general/src/lib/cfa/nsswitch.rb

```diff
@@ -0,0 +1,273 @@
+  class Nsswitch < BaseModel
+    include Yast::Logger
+
+    # Path to local configuration file
+    LOCAL_PATH = "/etc/nsswitch.conf".freeze
+    private_constant :LOCAL_PATH
+
+    # Path to vendor configuration file
+    VENDOR_PATH = "/usr/etc/nsswitch.conf".freeze
+    private_constant :VENDOR_PATH
+
+    class << self
+      # Instantiates and loads the file
+      #
+      # This method is basically a shortcut to instantiate and load the content in just one call.
+      #
+      # @param file_handler [#read,#write] something able to read/write a string (like File)
+      # @return [Nsswitch] File with the already loaded content
+      def load(file_handler: Yast::TargetFile)
+        new(file_handler: file_handler).tap(&:load)
+      end
+    end
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_handler: Yast::TargetFile)
+      super(AugeasParser.new("nsswitch.lns"), load_path, file_handler: file_handler)
+    end
+
+    # Loads the file content
+    #
+    # If the file does not exist, consider it as empty.
+    #
+    # @see CFA::BaseModel#load
+    def load
+      super
+      fix_collection_names(data)
+      @current_content = @parser.serialize(data)
+    rescue Errno::ENOENT # PATH does not exist yet
+      self.data = @parser.empty
+      @loaded = true
+    end
+
+    # Returns a list of defined databases
+    #
+    # @return [Array<String>] List of database names
+    def entries
+      databases.keys
+    end
+
+    # Service specifications for given database name
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    #
+    # @return [Array<String>] database service specifications
+    def services_for(db_name)
+      databases[db_name]&.services
+    end
+
+    # Update (or create if it does not exist yet) the entry for given database name
+    #
+    #
+    # @note Current services, if any, are completely replaced
+    #
+    # @see #entry_for
+    # @see DatabaseEntry#services=
+    #
+    # @param db_name [String] the database name, e.g., "passwd" or "hosts"
+    # @param services [Array<String>] service specifications for the database
+    def update_entry(db_name, services)
+      database = entry_for(db_name)
+      database.services = services
+    end
+
+    # Delete entry for given database name
+    #
+    # @param db_name [String] the database db_name, e.g., "passwd" or "hosts"
+    def delete_entry(db_name)
+      data.delete(database_matcher(db_name))
+    end
+
+    # Writes the current content to /etc/nsswitch.conf
+    #
+    # @see CFA::BaseModel#save
+    def save
+      return false unless modified?
+
+      new_content = @parser.serialize(data)
+
+      @parser.file_name = write_path if @parser.respond_to?(:file_name=)
+      @file_handler.write(write_path, new_content)
+      @current_content = new_content
+
+      true
```

Comment:
It is different then BaseModel save, where it returns void and here it is boolean if it is modified. What is reason?

Reply:
No, there is no strong reason. It was just because the use of `return false` made me think returning true when successfully. I'll change it to return `nil` when file not modified yet and keep the `@file_handler.write` line as the last line of the method. Although having a true/false as return value in a `#save` method looks reasonable to me.